lib/brainstem/api_docs/endpoint.rb

```diff
@@ -120,32 +120,61 @@ def valid_params
 
 
       #
-      # Returns a hash of all root-level params with values of an array of
-      # the parameters nested underneath, or +nil+ in the event they are
-      # root-level non-nested params.
-      #
-      # @return [Hash{Symbol => Array,NilClass}] root keys and the keys
-      #   nested under them, or nil if not a nested param.
-      #
-      def root_param_keys
-        @root_param_keys ||= begin
-          valid_params.to_h
-            .inject({}) do |hsh, (field_name, data)|
-              next hsh if data[:nodoc]
-
-              if data.has_key?(:root)
-                key  = data[:root].respond_to?(:call) ? data[:root].call(controller.const) : data[:root]
-                (hsh[key] ||= []) << field_name
+      # Returns a hash of all params nested under the specified root or
+      # parent fields along with their type, item type & children.
+      #
+      # @return [Hash{Symbol => Hash}] root keys and their type info, item info & children
+      #   nested under them.
+      #
+      def params_configuration_tree
```

Comment:
Following the logic for this method is pretty tedious. Is it possible to break it down further into more helper methods and lay down the general flow in a comment here -- does this make sense as an actionable revision?